### ISSUE TITLE
ButtonBar selected indicator buggy after calling moveToViewController

### ISSUE THREAD

**Issue by tterpell** [xmartlabs/XLPagerTabStrip#546]
Hello,

i call 
`override func viewDidAppear(_ animated: Bool) {
        moveToViewController(at: 1, animated: false)
    }`
in my ButtonBarPagerTabStripViewController. After doing this, the selected indicator animation is not working properly. In the attached video, u can see that the indicator is jumping around for no reason. E.g. If i click on Tab A, while being in tab B, the indicator jumps to tab C and then to B and finally to A.

[https://drive.google.com/file/d/15qtEtowgM7Ut1f4EfCbfIE43lSTxDSA_/view?usp=sharing](url)

What can i do to prevent this?

**Comment by OuSS-90** [comment]
I have same problem, any news how to fix?

**Comment by tterpell** [comment]
No. I used a Workaround, but i am on vacation and cant access my code atm.

On Mar 13, 2018 17:46, "OuSS-90" <notifications@github.com> wrote:

I have same problem, any news how to fix?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/xmartlabs/XLPagerTabStrip/issues/546#issuecomment-372712722>,
or mute the thread

**Comment by OuSS-90** [comment]
Ok thank you

**Comment by joserafaelrocha** [comment]
I'm also facing this issue, any news?

**Comment by joserafaelrocha** [comment]
After commenting 
this line ButtonBarPAgerTabStripViewController 196:
```swift
 buttonBarView.moveTo(index: currentIndex, animated: false, swipeDirection: .none, pagerScroll: .scrollOnlyIfOutOfScreen)
``` 

In my case, I do not face this bug anymore, didn't test other scenarios where it can create issues...

**Comment by tterpell** [comment]
Yes ofcourse, you can leave this line out. But that does not solve the problem.
My workaround is as follows:
At first, i call 
`buttonBarView.move(fromIndex: 0, toIndex: Int(selectedTab), progressPercentage: 1, pagerScroll: .no)`
after 
`moveToViewController(at: Int(selectedTab), animated: false)`
so that yields to 
`moveToViewController(at: Int(selectedTab), animated: false)
        buttonBarView.move(fromIndex: 0, toIndex: Int(selectedTab), progressPercentage: 1, pagerScroll: .no)`

Secondly, i overwrite the updateIndicator method with this 
`override func updateIndicator(for viewController: PagerTabStripViewController, fromIndex: Int, toIndex: Int, withProgressPercentage progressPercentage: CGFloat, indexWasChanged: Bool) {
        buttonBarView.move(fromIndex: 0, toIndex: toIndex, progressPercentage: 1, pagerScroll: .no)
    }`

I know thats not the solution we all wanted, but it works.

**Comment by ViennvNewBees** [comment]
Your status maybe sub controller just override (viewWillAppeal viewWillDidAppear, ...) but don't call (super.viewWillAppeal() ... ).
I have new problem with this method : after change to new sub controller but old controller status buttonBarView not change.
_1_ 2 3 4
_1_ 2 3 _4_

**Comment by tterpell** [comment]
For the originally reported bug, just override updateIndicator like this: 
```
override func updateIndicator(for viewController: PagerTabStripViewController, fromIndex: Int, toIndex: Int, withProgressPercentage progressPercentage: CGFloat, indexWasChanged: Bool) {
        buttonBarView.move(fromIndex: 0, toIndex: toIndex, progressPercentage: 1, pagerScroll: .no) 
    }
```

**Comment by watermamal** [comment]
> Your status maybe sub controller just override (viewWillAppeal viewWillDidAppear, ...) but don't call (super.viewWillAppeal() ... ).
> I have new problem with this method : after change to new sub controller but old controller status buttonBarView not change.
> **_1_** 2 3 4
> **_1_** 2 3 **_4_**

I have the same problem
